From de686a9e6d8c909485b87ca09c8f912bf83082f2 Mon Sep 17 00:00:00 2001
From: Damien Regad <dregad@mantisbt.org>
Date: Fri, 2 Feb 2018 12:14:42 +0100
Subject: [PATCH] Fix PHP error - wrong argument type

Initialize $t_filter variable as array() instead of '' in
current_user_get_bug_filter(), to ensure its type is correct when
calling filter_ensure_valid_filter().

Fixes #23921
---
 core/current_user_api.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/current_user_api.php b/core/current_user_api.php
index 85f854b833..e60e00504d 100644
--- a/core/current_user_api.php
+++ b/core/current_user_api.php
@@ -238,7 +238,7 @@ function current_user_ensure_unprotected() {
  */
 function current_user_get_bug_filter( $p_project_id = null ) {
 	$f_filter_string = gpc_get_string( 'filter', '' );
-	$t_filter = '';
+	$t_filter = array();
 
 	if( !is_blank( $f_filter_string ) ) {
 		if( is_numeric( $f_filter_string ) ) {
